mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLXMLClassificationManager.java

```diff
@@ -0,0 +1,213 @@
+
+/**
+ * OCFL File Manager for MyCoRe Classifications
+ * @author Tobias Lenhardt [Hammer1279]
+ */
+public class MCROCFLXMLClassificationManager implements MCRXMLClassificationManager {
+
+    protected static final String CLASSIFICATION_PREFIX = "mcrclass:";
+
+    private static final boolean INC_CLSDIR = MCRConfiguration2.getBoolean(CLASSIFICATION_PREFIX).orElse(false);
+
+    public static final String MESSAGE_CREATED = "Created";
+
+    public static final String MESSAGE_UPDATED = "Updated";
+
+    public static final String MESSAGE_DELETED = "Deleted";
+
+    private String rootFolder = INC_CLSDIR ? "classification/" : "";
+
+    @MCRProperty(name = "Repository", required = true)
+    public String repositoryKey;
+
+    protected static final Map<String, Character> MESSAGE_TYPE_MAPPING = Collections.unmodifiableMap(Map.ofEntries(
+        Map.entry(MESSAGE_CREATED, MCROCFLMetadataVersion.CREATED),
+        Map.entry(MESSAGE_UPDATED, MCROCFLMetadataVersion.UPDATED),
+        Map.entry(MESSAGE_DELETED, MCROCFLMetadataVersion.DELETED)));
+
+    protected static char convertMessageToType(String message) throws MCRPersistenceException {
+        if (!MESSAGE_TYPE_MAPPING.containsKey(message)) {
+            throw new MCRPersistenceException("Cannot identify version type from message '" + message + "'");
+        }
+        return MESSAGE_TYPE_MAPPING.get(message);
+    }
+
+    protected OcflRepository getRepository() throws ClassCastException {
+        return MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    }
+
+    public void create(MCRCategory mcrCg, MCRContent xml) {
+        fileUpdate(mcrCg, xml, MESSAGE_CREATED);
+    }
+
+    public void update(MCRCategory mcrCg, MCRContent xml) {
+        fileUpdate(mcrCg, xml, MESSAGE_UPDATED);
+    }
+
+    void fileUpdate(MCRCategory mcrCg, MCRContent xml, String messageOpt) {
+        
+        String objName = getName(mcrCg.getId());
+        Date lastModified = new Date(MCRCategoryDAOFactory.getInstance().getLastModified(mcrCg.getId().getID()));
+        String message = messageOpt; // PMD Fix - AvoidReassigningParameters
+        if(Objects.isNull(message)) {
+            message = MESSAGE_UPDATED;
+        }
+
+        try (InputStream objectAsStream = xml.getInputStream()) {
+            VersionInfo versionInfo = buildVersionInfo(message, lastModified);
+            getRepository().updateObject(ObjectVersionId.head(objName), versionInfo, updater -> {
+                updater.writeFile(objectAsStream, buildFilePath(mcrCg), OcflOption.OVERWRITE);
+            });
+        } catch (IOException e) {
+            throw new MCRPersistenceException("Failed to update object '" + objName + "'", e);
+        } catch (IllegalArgumentException e) {
+            // Issue seems to be with the OCFL Library trying to find a file it deleted itself
+            // LOGGER.error("Something has gone Wrong!", e);
+        }
+
+    }
+
+    public void fileDelete(MCRCategoryID mcrid) {
```

Comment:
method naming: We have "create", "update" and "fileDelete". Should be renamed to "delete"